src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -941,6 +972,16 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
             if new_schema:
                 schema = new_schema
 
+        if self._user_schema and not self._infer_schema and format == "JSON":
```

Comment:
do we need check `format == "JSON"` here since it's already checked in `_infer_schema_from_user_input`?
also I'm curious what would happen if users pass non-JSON format

Reply:
This is the part that I am still working on, spark will not infer schema and accept user schema at the same time, I am working on to have it in snowpark

Reply:
if non-json format is received, the behavior is the same as before